src/tools/std23/views/chunk.hpp

```diff
@@ -0,0 +1,374 @@
+               std::enable_if_t< std20::same_as< Iterator, iterator< ! Const > >, int > = 0,
+               bool C = Const, typename RR = R,
+               std::enable_if_t< C && std20::convertible_to< std20::ranges::iterator_t< RR >,
+                                                             std20::ranges::iterator_t< Base > >, int > = 0 >
+    constexpr iterator( Iterator iter ) :
+        current_( std::move( iter.current_ ) ),
+        end_( std::move( iter.end_ ) ),
+        n_( iter.n_ ),
+        missing_( iter.missing_ ) {}
+
+    constexpr std20::ranges::iterator_t< Base > base() const { return this->current_; }
+
+    constexpr value_type operator*() const {
+
+      return std20::views::take( std20::ranges::subrange( this->current_, this->end_ ), this->n_ );
+    }
+
+    constexpr iterator& operator++() {
+
+      this->missing_ = std20::ranges::advance( this->current_, this->n_, this->end_ );
+      return *this;
+    }
+
+    constexpr iterator operator++(int) {
+
+      auto temp = *this;
+      ++*this;
+      return temp;
+    }
+
+    template < typename B = Base >
+    constexpr auto operator--()
+    -> std::enable_if_t< std20::ranges::bidirectional_range< B >, iterator& > {
+
+      std20::ranges::advance( this->current_, this->missing_ - this->n_ );
+      this->missing_ = 0;
+      return *this;
+    }
+
+    template < typename B = Base >
+    constexpr auto operator--(int)
+    -> std::enable_if_t< std20::ranges::bidirectional_range< B >, iterator > {
+
+      auto temp = *this;
+      --*this;
+      return temp;
+    }
+
+    template < typename B = Base >
+    constexpr auto operator+=( difference_type n )
+    -> std::enable_if_t< std20::ranges::random_access_range< B >, iterator& > {
+
+      if ( n > 0 ) {
+
+        std20::ranges::advance( this->current_, this->n_ * ( n - 1 ) );
+        this->missing_ = std20::ranges::advance( this->current_, this->n_, this->end_ );
+      }
+      else if ( n < 0 ) {
+
+        std20::ranges::advance( this->current_, this->n_ * n + this->missing_ );
+        this->missing_ = 0;
+      }
+      return *this;
+    }
+
+    template < typename B = Base >
+    constexpr auto operator-=( difference_type n )
+    -> std::enable_if_t< std20::ranges::random_access_range< B >, iterator& > {
+
+      return *this += -n;
+    }
+
+    // this version of operator[] does not work on gcc
+    // template < typename B = Base >
+    // constexpr auto operator[]( difference_type n ) const
+    // -> std::enable_if_t< std20::ranges::random_access_range< B >, value_type > {
+    //
+    //   return *( *this + n );
+    // }
+
+    template < typename B = Base,
+               typename = std::enable_if_t< std20::ranges::random_access_range< B > > >
+    constexpr decltype(auto) operator[]( difference_type n ) const {
+
+      return *( *this + n );
+    }
+
+    template < typename B = Base >
+    friend constexpr auto operator==( const iterator& left, const iterator& right )
```

Comment:
Yes, these do not necessarily need to be friends. It's more or less a question of style I think. 

```
friend constexpr auto operator==( const iterator& left, const iterator& right ) {

  return left.current_ == right.current_;
}
```
or 
```
constexpr auto operator==( const iterator& right ) {

  return this->current_ == right.current_;
}
```
are in effect the same.

I basically use the friends version here because it was done this way in nanorange too (the std20 namespace in tools is a slimmed down and modified version of nanorange), as you can see here:
https://github.com/njoy/tools/blob/main/src/tools/std20/views/transform.hpp